# Project 2: E-Commerce Business Dashboard
## Dataset: Brazilian E-Commerce Public Dataset by Olist
### 5 Business Questions This Project Answers
1. What are the overall KPIs: total revenue, total orders, average order value, return rate?
2. How does revenue grow month over month and what is the percentage change?
3. Which product categories drive the most revenue and which are declining?
4. Which customer geography contributes most to sales?
5. How do different customer segments compare on order value and frequency?

In [1]:
import pandas as pd

In [7]:
orders    =pd.read_csv('../data/raw/olist_orders_dataset.csv')
items     = pd.read_csv('../data/raw/olist_order_items_dataset.csv')
payments  = pd.read_csv('../data/raw/olist_order_payments_dataset.csv')
customers = pd.read_csv('../data/raw/olist_customers_dataset.csv')
products  = pd.read_csv('../data/raw/olist_products_dataset.csv')

In [10]:
# Inspect each table
for name, df in [('orders', orders),
                 ('items', items),
                 ('payments', payments),
                 ('customers', customers),
                 ('products', products)]:
    print(f'\n{"="*40}')
    print(f'TABLE: {name}')
    print(f'Shape: {df.shape}')
    print(f'\nColumns: {df.columns.tolist()}')
    print(f'\nNull values:\n{df.isnull().sum()}')


TABLE: orders
Shape: (99441, 8)

Columns: ['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp', 'order_approved_at', 'order_delivered_carrier_date', 'order_delivered_customer_date', 'order_estimated_delivery_date']

Null values:
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

TABLE: items
Shape: (112650, 7)

Columns: ['order_id', 'order_item_id', 'product_id', 'seller_id', 'shipping_limit_date', 'price', 'freight_value']

Null values:
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

TABLE: payments
Shape: (103886, 5)

Columns: ['order_id', 'payment_sequential', 'pa

## Task 1 Findings - Data Inspection

### Table Sizes
- orders: 99,441 rows - one row per order
- items: 112,650 rows  multiple items per order
- payments: 103,886 rows — multiple payment methods per order
- customers: 99,441 rows — matches orders exactly
- products: 32,951 rows — one row per product

### Key Issues Found
- orders: 2,965 orders never delivered (null delivery date)
- orders: 160 orders never got payment approval
- products: 610 products missing category name

### Table Connections
- orders connects to customers via customer_id
- orders connects to payments via order_id
- orders connects to items via order_id
- items connects to products via product_id